File: mgyoutube-multitech/api-go/src/repos/userdatarepo.go
package repos

import "apimodel"

// UserDataRepo - interface definition for a UserDataRepo
type UserDataRepo interface {
	RepositoryStartup() error

	GetUserByUsername(username string) *apimodel.User

	AddUser(user *apimodel.User) *apimodel.User

	RemoveUser(user *apimodel.User)

	ReplaceUser(userID string, user *apimodel.User) *apimodel.User

	AddChildToParent(parentUserID string, childUserID string)

	GetChildrenForParent(parentUserID string) []*apimodel.User
}

File: mgyoutube-multitech/api-java/src/main/java/com/djpedesen/mgyoutube/api_java/repos/UserDataRepo.java
package com.djpedesen.mgyoutube.api_java.repos;

import java.util.List;

import com.djpedesen.mgyoutube.api_java.apimodel.User;

public interface UserDataRepo {
	void repositoryStartup() throws Exception;

	User getUserByUsername(String username) throws Exception;

	User addUser(User user) throws Exception;

	void removeUser(User user) throws Exception;

	User replaceUser(String userId, User user) throws Exception;

	void addChildToParent(String parentUserId, String childUserId) throws Exception;

	List<User> getChildrenForParent(String parentUserId) throws Exception;

}
